tests/test_int_penaltymodel_ecosystem.py

```diff
@@ -1,38 +1,44 @@
 import unittest
+from unittest import mock
+import tempfile
 
 import networkx as nx
 
 import dimod
 
 import penaltymodel.core as pm
-import penaltymodel.cache as pmc
-import penaltymodel.maxgap as maxgap
-
 
 class TestInterfaceWithCache(unittest.TestCase):
     def test_retrieval(self):
-        # put some stuff in the database
+        tmp_db_file = tempfile.NamedTemporaryFile().name
 
-        spec = pm.Specification(nx.path_graph(2), (0, 1), {(-1, -1), (1, 1)}, vartype=pm.SPIN)
-        model = dimod.BinaryQuadraticModel({0: 0, 1: 0}, {(0, 1): -1}, 0.0, vartype=pm.SPIN)
-        widget = pm.PenaltyModel.from_specification(spec, model, 2, -1)
+        with mock.patch("penaltymodel.cache.database_manager.cache_file", lambda: tmp_db_file):
```

Comment:
It's an overkill here, but it would be nice to implement this pattern as a decorator `@isolated_cache`.

Reply:
I've implemented the decorator, but not sure where I should put it.

Reply:
Probably makes sense to be a utility in `penaltymodel-cache`. Also, check out [functools.wraps](https://docs.python.org/3.8/library/functools.html#functools.wraps).